lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -1948,6 +1978,43 @@ impl InstanceDef {
 
 
 
+// ========================
+// === Layer assignment ===
+// ========================
+
+/// Identifies an assigned layer, including virtual sublayer information, if any.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct LayerAssignment {
+    /// The layer.
+    pub layer:            WeakLayer,
+    /// The virtual sublayer, if any.
+    pub virtual_sublayer: Option<VirtualSublayerAssignment>,
+}
+
+
+// === Virtual sublayers ===
+
+/// Identifies an assigned virtual sublayer.
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct VirtualSublayerAssignment {
+    /// The shape system the virtual sublayer applies to.
+    pub shape_system: crate::display::shape::ShapeSystemId,
+    /// The virtual sublayer ID.
+    pub virtual_id:   VirtualLayerId,
+}
```

Comment:
Ok, here I have another very minor naming problem. `LayerAssignment` contains assignment to both layer and sublyaer, while `VirtualSublayerAssignment` contains assignment only for sublayer. It's not obvious IMO from these names, do you agree with that?